mlir/include/mlir/Dialect/ArmSME/IR/ArmSMEOps.td

```diff
@@ -60,6 +60,12 @@ def TileElementWidthMatchesTileID : TypesMatchWith<
                   "::llvm::cast<VectorType>($_self).getElementType())"
                   ".getWidth())">;
 
+class HasMatchingMaskTypeConstraint<string vector, string mask> :
+  OptionalTypesMatchWith<
+    mask # " has i1 element type and same shape as " # vector,
+    vector, mask,
+    "::llvm::cast<mlir::VectorType>($_self).cloneWith({}, IntegerType::get($_ctxt, 1))">;
```

Comment:
@MacDue I've based on the constraint you're adding in #69604 but parameterized it

Reply:
You should be able to rebase and de-duplicate this now :)

Reply:
Done :)